master/buildbot/test/unit/test_process_buildrequest.py

```diff
@@ -354,18 +359,39 @@ def test_cancelBuildRequest(self):
                                   fakedb.SourceStamp(id=1, sourcestampsetid=1)])
 
         d = master.db.buildrequests.getBuildRequest(1)
+        d.addCallback(lambda brdict:
+                      buildrequest.BuildRequest.fromBrdict(master, brdict))
+        d.addCallback(lambda br: br.cancelBuildRequest())
+        d.addCallback(lambda _: master.db.buildrequests.getBuildRequest(1))
+        d.addCallback(checkCanceled)
```

Comment:
Minor: Maybe it would be little more pretty if we will use a lambda expression in this way:
```
d.addCallback(lambda brdict: self.assertEqual(brdict['results'], CANCELED)
d.addCallback(lambda brdict: self.assertEqual(brdict['complete'], 1)
```

Reply:
In this case I've used self.assertDictContainsSubset and remove an inner function.